src/Sarif/Visitors/PerRunPerRuleSplittingVisitor.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class PerRunPerRuleSplittingVisitor : SarifRewritingVisitor
+    {
+        public PerRunPerRuleSplittingVisitor(Func<Result, bool> filteringStrategy = null)
```

Comment:
>PerRunPerRuleSplittingVisitor [](start = 15, length = 29)

i wonder if the word "perRun" is adding any value here? wouldn't the bugs be logged always on the per run basis? (assuming the filtering strategy filters out the new ones)

Reply:
you could imagine a single bug per log file. imagine a complete SDL run against a small project, where you file a single bucket bug. also, a log file could have multiple runs of the same tool, which could be aggregated. 

---
In reply to: [308476793](https://github.com/microsoft/sarif-sdk/pull/1596#discussion_r308476793) [](ancestors = 308476793)